70-89/70-89.md

```diff
@@ -1,9 +1,744 @@
-#### Decoupling 70
+#### Decoupling
 
-##### Struct Composition 70
+##### Struct Composition 
 
-##### Decoupling With Interface 74
+
+Prototyping is important, as well as writing proof of concept and solving problems in the concrete first. Then we can ask ourselves: What can change? What change is coming? so we can start decoupling and refactor. Refactoring needs to become a part of the development cycle.
+
+Here is the problem that we are trying to solve in this section. We have a system called Xenia that has a database. There is another system called Pillar, which is a web server with some front-end that consumes it. It has a database too. Our goal is to move Xenia's data into Pillar's system.
+
+How long is it gonna take? How do we know when a piece of code is done so we can move on the next piece of code? If you are a technical manager, how do you know whether your debt is "wasting effort" or "not putting enough effort"?
+
+To answer, being done has 2 parts. One is test coverage, 80% in general and 100% on the happy path. Second is about changes. By asking what can change, from a technical perspective and business perspective, we make sure that we refactor the code to be able to handle that change.
+
+One example is, we can give you a concrete version in 2 days but we need 2 weeks to be able to refactor this code to deal with the change that we know it's coming. The plan is to solve one problem at a time. Don't be overwhelmed by everything. Write a little code, write some tests and refactor. Write a layer of APIs that work on top of each other, knowing that each layer is a strong foundation to the next.
+
+Do not pay too much attention to the implementation detail. It's the mechanics here that are important. We are optimizing for correctness, not performance. We can always go back if it doesn't perform well enough to speed things up.
+
+```go
+package main
+
+import ( 
+	"errors"
+	"fmt" 
+	"io"
+	"math/rand"
+	"time"
+)
+```
+
+The first problem that we have to solve is that we need software that runs on a timer. It needs to connect to Xenia, read that database, identify all the data we haven't moved and pull it in.
+
+```go
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+```
+
+Data is the structure of the data we are copying. For simplicity, just pretend it is a string data.
+
+```go
+type Data struct { 
+	Line string
+}
+```
+
+Xenia is a system we need to pull data from.
+
+```go
+type Xenia struct { 
+	Host string
+	Timeout time.Duration 
+}
+```
```

Comment:
```suggestion
```go
type Data struct { 
    Line string
}
```

Xenia is a system we need to pull data from.

```go
type Xenia struct { 
    Host string
    Timeout time.Duration 
}
```
```

indent to 4